### PR TITLE
feat: improve provider fetching

### PR DESCRIPTION
**Pull request by dignifiedquire**
Integrates https://github.com/libp2p/rust-libp2p/pull/2712 to fetch providers more quickly. For widely available providers this reduces the fetch time considerably.

### CODE REVIEW

**Review by Arqu — APPROVED** [review]
Looking forward to this one, if it just resolves most of the 0 provider requests we're ~10% better off than before.

File: iroh-p2p/src/service.rs [hunk 1]
```diff
@@ -86,7 +89,7 @@ impl Libp2pService {
             .with_max_pending_outgoing(Some(30)) // TODO: configurable
             .with_max_established_incoming(Some(config.target_peer_count))
             .with_max_established_outgoing(Some(config.target_peer_count))
-            .with_max_established_per_peer(Some(5)); // TODO: configurable
+            .with_max_established_per_peer(Some(60)); // TODO: configurable
```
[the remark this answers is not in the document]
Comment (dignifiedquire) [reply iroh-p2p/src/service.rs:92]: wrong number I changed 😓

**Comment by Arqu** [comment]
Tested locally, after some warmup time/requests, things seem a lot better than before.

**Review by Arqu — APPROVED** [review]
⚡

**Pull request merged**